### ISSUE TITLE
Union of `T` and `Any` doesn't conform to `T` (`T` is a type variable)

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#2054]
## Environment data

-   Language Server version: 2021.11.1-pre.1
-   OS and version: Window 7 x64
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.12 Anaconda

## Sample code

```python
from __future__ import annotations

from typing import Callable, TypeVar

T = TypeVar("T")

def get_value(key: str, gen: Callable[[], T]) -> T:
    try:
        value = globals()[key]  # Any
    except KeyError:
        value = gen()  # T
    return value  # Any | T
```

## Expected behavior

No warning.

## Actual behavior

![image](https://user-images.githubusercontent.com/15389744/140870022-4768c447-5a20-4058-bbaa-1eed04900f2b.png)

# Additional info

There would be no warning if the type of `value` were either `T` or `Any` instead of a union.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by heejaechang** [comment]
fixed in 2021.11.1 release
